src/autoscaler/metricscollector/config/config.go

```diff
@@ -126,6 +132,9 @@ func (c *Config) Validate() error {
 		return fmt.Errorf("Configuration error: RefreshInterval is 0")
 	}
 
+	if (c.Collector.CollectMethod != CollectMethodPolling) && (c.Collector.CollectMethod != CollectMethodStreaming) {
```

Comment:
if we expect more collect methods in future, then maybe can use an enum to avoid adding conditions here.